### PR TITLE
refactor(pie-switch): DSW-1582 use form control mixin

### PR DESCRIPTION
**Pull request by jamieomaguire**
Tested on PIE Aperture NextJS (SSR) & Vanilla App

---
"@justeattakeaway/pie-switch": minor
---

[Changed] - Use new FormControlMixin for form association logic in `pie-switch` component

---
"@justeattakeaway/pie-webc-core": minor
---

[Changed] - Use proper interface keyword for RTLmixin interface. Was using a class before which is incorrect.

---
"@justeattakeaway/pie-webc-core": minor
---

[Removed] - unused LitElementMixin type as we do not need it anywhere

### CODE REVIEW

File: packages/components/pie-webc-core/src/mixins/rtl/rtlMixin.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import type { GenericConstructor } from '../types/GenericConstructor';
  * An interface representing the structure of RTL related class.
  * @interface
  */
-declare class _RTLInterface {
+export interface RTLInterface {
```
Comment (raoufswe) [inline packages/components/pie-webc-core/src/mixins/rtl/rtlMixin.ts:9]: 🔥

**Review by raoufswe — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**